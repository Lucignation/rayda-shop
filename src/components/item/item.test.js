import Item from "./item.component";
import renderer from "react-test-renderer";

const _product = {
  title: "title",
  name: "Gerald Olumide",
  image: "https://i.ibb.co/h7Bts6w/Imagelaptop.png",
  bid: "â‚¦795,000",
};

const {name} = _product;

const nickNameArr = name.split(" ");
const nickname = `${nickNameArr[0][0]}${nickNameArr[1][0]}`; 

const mockProps = {
  product: _product,
};

describe("Product", () => {
  it("renders a snapshot of <Item /> when passed into a component", () => {
    const product = renderer.create(<Item {...mockProps} />).toJSON();

    expect(product).toMatchSnapshot();
  });
});
