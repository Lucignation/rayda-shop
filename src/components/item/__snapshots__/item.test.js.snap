// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders a snapshot of <Item /> when passed into a component 1`] = `
<div
  className="item-section"
>
  <div
    className="item-featured-image"
  >
    <img
      alt="featured avatar"
      src="https://i.dell.com/is/image/DellContent/content/dam/ss2/product-images/page/category/brand-compare/cat-page-brand-compare-xps-13-9320-sl-front-800x550.png?fmt=png-alpha&wid=800&hei=550"
    />
  </div>
  <div
    className="item-bidder"
  >
    <h3
      className="item-bidder-nickname"
    />
    <h4
      className="item-bidder-name"
    >
       
      <span
        className="item-highest-bidder"
      >
        (Highest Bidder)
      </span>
    </h4>
  </div>
  <div
    className="item-title"
  />
  <div
    className="item-price"
  >
    <span
      className="item-price-title"
    >
      Current Bid:
    </span>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18opu5d-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Add to wishlist
  </button>
</div>
`;
